Sources/ArcGISToolkit/Components/AR/TableTopSceneView.swift

```diff
@@ -0,0 +1,241 @@
+                            .attributionBarHidden(true)
+                            .spaceEffect(.transparent)
+                            .viewDrawingMode(.manual)
+                            .atmosphereEffect(.off)
+                            .onAppear {
+                                self.sceneViewProxy = proxy
+                            }
+                            .opacity(didSetTransforamtion ? 1 : 0)
+                    }
+                }
+            }
+    }
+    
+    /// Updates the last good device orientation.
+    func updateLastGoodDeviceOrientation() {
+        // Get the device orientation, but don't allow non-landscape/portrait values.
+        let deviceOrientation = UIDevice.current.orientation
+        if deviceOrientation.isValidInterfaceOrientation {
+            lastGoodDeviceOrientation = deviceOrientation
+        }
+    }
+    
+    func visualizePlane(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
+        // Place content only for anchors found by plane detection.
+        guard let planeAnchor = anchor as? ARPlaneAnchor else { return }
+        
+        // Create a custom object to visualize the plane geometry and extent.
+        let plane = Plane(anchor: planeAnchor)
+        
+        // Add the visualization to the ARKit-managed node so that it tracks
+        // changes in the plane anchor as plane estimation continues.
+        node.addChildNode(plane)
+    }
+    
+    func updatePlane(_ renderer: SCNSceneRenderer, didUpdate node: SCNNode, for anchor: ARAnchor) {
+        if didSetTransforamtion {
+           node.removeFromParentNode()
+        }
+    
+        guard let planeAnchor = anchor as? ARPlaneAnchor,
+              let plane = node.childNodes.first as? Plane
+        else { return }
+        
+        // Update extent visualization to the anchor's new bounding rectangle.
+        if let extentGeometry = plane.node.geometry as? SCNPlane {
+            extentGeometry.width = CGFloat(planeAnchor.extent.x)
+            extentGeometry.height = CGFloat(planeAnchor.extent.z)
+            plane.node.simdPosition = planeAnchor.center
+        }
+    }
+}
+
+/// Helper class to visualize a plane found by ARKit
+class Plane: SCNNode {
+    let node: SCNNode
+    
+    init(anchor: ARPlaneAnchor) {
+        // Create a node to visualize the plane's bounding rectangle.
+        let extent = SCNPlane(width: CGFloat(anchor.extent.x), height: CGFloat(anchor.extent.z))
+        node = SCNNode(geometry: extent)
+        node.simdPosition = anchor.center
+        
+        // `SCNPlane` is vertically oriented in its local coordinate space, so
+        // rotate it to match the orientation of `ARPlaneAnchor`.
+        node.eulerAngles.x = -.pi / 2
+
+        super.init()
+
+        node.opacity = 0.25
+        guard let material = node.geometry?.firstMaterial
+            else { fatalError("SCNPlane always has one material") }
```

Comment:
if there is always a geometry and always a first material then it should be force unwrapped (`!`). If there are cases where geometry or firstMaterial are `nil` then it should be `guard let else { return `nil` }` and make the init failable.